### ISSUE TITLE
Error when copying instances of Generic subclasses

### ISSUE THREAD

**Issue by izquierdo** [python/typing#498]
Trying to do the following

```python
import copy
from typing import Generic, TypeVar

T = TypeVar('T')

class Test(Generic[T]):
    def __init__(self, t: T) -> None:
        self.t = t

test: Test[int] = Test(5)
copy.copy(test)
```

produces

```
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    copy.copy(test)
  File "/usr/lib/python3.6/copy.py", line 88, in copy
    return copier(x)
TypeError: __copy__() takes 1 positional argument but 2 were given
```

whereas there is no error if not inheriting from `Generic`.

The most similar issue to this one is #458 but that snippet is trying to call `__copy__` directly instead of `copy.copy`. There is also #306 but that one is trying to copy the type instead of an instance, and also switching the above snippet to `copy.deepcopy` does work. Finally, there is #468, it's possible that a fix for this issue would fall within its scope.

**Comment by ilevkivskyi** [comment]
Indeed, this is fixed by #468 (a.k.a. PEP 560), but the fix will be available only in Python 3.7. Therefore, I think it makes sense to also fix it in the backport (the crash also happens on Python 3.4).

**Comment by hynekcer** [comment]
A solution that works:
``` python
class MyClass(Generic[T]):
    __copy__ = None
    ...
```
This is safe:  
1) The value `__copy__ = None` has the same effect for the function copy.copy() like an undefined `__copy__` attribute. A possible problem is if it would overshadow a right `__copy__` attribute.
2) The "Generic" type is the last ancestor of MyClass. (The only possible problem would be if it is used as generic mixin for a class with a `__copy__` method, but it can be easily solved by a `super(...).__copy__()`)

A general hack can be also
```
from typing import Generic

if hasattr(Generic, '__copy__') and sys.version_info[:2] == (3, 5):
    delattr(Generic, '__copy__')
```
It works in some of the first modules of an application before the first copy.

Maybe this is also possible to do in typing_extensions.
